### DISCUSSION TITLE
Problem with grammar

### DISCUSSION THREAD

**Discussion by mnaderhirn** [eclipse-langium/langium#1661] [General]
I use this grammar like in this [playground](https://langium.org/playground?grammar=PTAEEkFsAcHsCcAuBLApvVoAmbQFVwARAWkNQDNkA7ZFWK0AQwFcBnbdUAUUdYE9WjcqmIBxeI0iRGKANYAoAOYSpjeKADCG6NHGTp8eUZAQqidOUYBjTOQA-6rKgYBpWFOeI186ufiWbTW1oUABveVBI0FYvJAB%2BAC4gnQBtAF0Abgio5yxE5Oh0rKjQZmQ8pJj4akUsgF9jMAAJFmhzXwlFVmhmKllEUHtHXD1VOXlPeD4C0AxEZngqdi0dBOzImLVEAGoAXgBlWMQAJQp0Z0CAH1Bcva4qLFPhDCobLPk-SGpGABsIQmOXAAiklgClGMQAF4AQWIAC0AAzEACcaVCAGYADQAVjqAAoADrEAlYUIAFjqAEpQgBGTEU4AZSLyAAW5ScDE%2B3z%2B%2BwACtCNFx9qCUqACYg0ttGaBWeznKAuVRfqAAHJcADqABkRQT4HECVRJYyPugvkqeQAVaHHC0AfX2wLwXBVgqSAHICcBPaAAAKoXh8FJuy4er3AP0BoNZNlYDkK03c0D7TW2jQAeQAshnnRbQZ7PSkAHoG3VpABUxsVyudhFtgIAYlxAS6uO7PW6slW-jX7Y7na7QG60h2TfAzcqMyn01mc3mvWWUgTWAT9uX9WXPZWE%2BbNGnCK3QGCy7aITD4WlFwB3W3l41GQ5bJ7nV6YOYLJYFNYlfZWm29oFOi2pTlLsRCAkC3AqrWDoAf2XDvPcjxnC8gRvosyzBF%2BUQ-tadowYBgqQbWDZNnBwFYKBALAkR-4EfBRhGEAA&content=GISQMgogBAVAZgBwC4CcDcAoDB6bUACApgIYDOAngNoBKEAigLQAMTAjEwLo54DyChAOyjEocAJYAbQlDFCUJACZQAtgHsFhDIlRQAvKNX8BACgBEE1fOViEpAK7KAdEgAeSUwBooplKYCUmLgEJBSU2LSMLOxc3FAACiiySFCkqsrSSIRuMnBQSAAW0uJSUAqqhKRQAqrJWWKkSLFEZFQRzEwATJwYYnDG2ih6%2BgByAKpgYH5QAN4YUFAIiQJIfabDNcIARiVIqlCGgnmFopKEjv6YAL5NIVTh9O1dMUEAwhakGcfFmnAAxu%2BEfrIFABKBAA)

What i dont understand, why it is not capturing the code lines?

**Comment by msujew** [comment]
Hey @mnaderhirn,

perhaps the playground link is missing something? There doesn't seem to be a parser rule to parse code lines.

**Reply by mnaderhirn** [reply]
now with a code line rule, still the same error
[playground](https://langium.org/playground?grammar=PTAEEkFsAcHsCcAuBLApvVoAmbQFVwARAWkNQDNkA7ZFWK0AQwFcBnbdUAUUdYE9WjcqmIBxeI0iRGKANYAoAOYSpjeKADCG6NHGTp8efJCgAEi2iJU1RBMWtozKrMShyAH-U5Me1XPmoVLZ8mtrQAFzyoNGgrIhqiADUALwAyvFIAEoU6IEAxpgAPqCBWClcVFjZwhhUBaDFebBYqMkazagAMtSoANxGVvCQ1IwANhCEmVwAiuGgwADajMQAXgCCxABaAAzEAJwAugDeAMwANACsAL4AFAA6xHdYRwAsVwCURwCMZ2-AvdF5AALZBYFoMQbDKhjUCpAAKaw0XFSc0WoDuiAOiX%2BoGBoPBoEhI3GADkuAB1TqohZ3eAAfjuVCx-3kROh41SABU1plOQB9VIzPBcElIuYAcjuwCloAAAqheHwFuLCpLpcB5Yrlf0QWDAoT0FCYalOnyNAB5ACylpFnNRUqlCwAeozaQcAFQstkwkWEPlTABiXCmoq4Eql4v63vGvoFQpFYtA4oOkdZhuJoEtpot1tt9ul7pprDuqQ9DPdUq96fZE2p7r5y3WWwONIA7nyPSz5O0Wt0qKhIjFQE0WikiKAbiPUGPCO93f15OkEtVcnUB1EYlyefzBdNhaHQMxQckiFNptwSX7d-ukQuKlUcrUCoPN9zeXG9wmuBe-YHg1-D2PU8Zh-D8by4BcjCAA&content=GISQMgogBAVAZgBwC4CcDcAoDB6bUACApgIYDOAngNoBKEAigLQAMTAjEwLo54DyChAOyjEocAJYAbQlDFCUJACZQAtgHsFhDIlRQAvKNX8BACgBEE1fOViEpAK7KAdEgAeSUwBooplKYCUmLgEJBSU2LSMLOxc3FAACiiySFCkqsrSSIRuMnBQSAAW0uJSUAqqhKRQAqrJWWKkSLFEZFQRzEwATJwYYnDG2ih6%2BgByAKpgYH5QAN4YUFAIiQJIfabDNcIARiVIqlCGgnmFopKEjv6YAL5NIVTh9O1dMUEAwhakGcfFmnAAxu%2BEfrIFABKBAA)

**Reply by msujew** [reply]
Ok, but most of the code still isn't parsing. I.e. all expression rules are missing. Note that the unexpected way of defining hidden token broke the comment textmate grammar generator. See [here](https://langium.org/playground?grammar=PTAEEkFsAcHsCcAuBLApvVoAmbQFVwARAWkNQDNkA7ZFWK0AQwFcBnbdUAUUdYE9WjcqmIBxeI0iRGKANYAoAOYSpjeKADCG6NHGTp8efJCgAEi2iJU1RBMWtozKrMShyAH-U5Me1XPmoVLZ8mtrQAFzyoNGgABSsiGqIANQAvADKiUgAShTogQDGmAA%2BoIFYaVxUWLnCGFRFoKUFsFioqRqtqAAy1KgAlABUANxGVvCQ1IwANhCE2VwAiuGgwADajMQAXgCCxABaAAzEAJwAugDeAMwANACsAL6xADrEz1gXACwP-RcAjDdvsBhtF5AALZBYNoMcaTKgzUDpAAKOw0XHSK3Wz1YZ2SwNA8lhU1m6QAKjtsqSAPrpJZ4LgAOTRKwA5M9gOzQAABVC8PhrFnFNkc4A8vkC0YQqGBUBE%2BEk7pUjQAeQAsqrGaTMez2WsAHrPKjPeBnQbAwnoOEIxmEKkLABiXAWTK4rPZLNGcutDNttMW9JdrLOHotE2JoFVipV6s12o5gzW2Oe6VNAH5noN2eavbMiJi1oMqZtdgczomAO5U03m%2BSdNq9KioSIxUAtNppIhxNuoDuEIagD0h%2BSZJK1fINJtRGJkinUv0BtGgZiQ1JEBaLbg%2Bml0xlo0byKo1PL1IrN6fkynb-27rib20Op03pcrtdLO9XhdcfdGIA&content=GISQMgogBAVAZgBwC4CcDcAoDB6bUACApgIYDOAngNoBKEAigLQAMTAjEwLo54DyChAOyjEocAJYAbQlDFCUJACZQAtgHsFhDIlRQAvKNX8BACgBEE1fOViEpAK7KAdEgAeSUwBooplKYCUmLgEJBSU2LSMLOxc3FAACiiySFCkqsrSSIRuMnBQSAAW0uJSUAqqhKRQAqrJWWKkSLFEZFQRzEwATJwYYnDG2ih6%2BgByAKpgYH5QAN4YUFAIiQJIfabDNcIARiVIqlCGgnmFopKEjv6YAL5NIVTh9O1dMUEAwhakGcfFmnAAxu%2BEfrIFABKBAA) for something that works (a bit) better.
